### PR TITLE
[SCClassLibrary] Do not load Qt dependent classes on Qt-less builds

### LINKED ISSUE #1209: Primitive not bound for QtGui when running on a headless system (readline)
Supposedly this was fixed in:
https://github.com/supercollider/supercollider/commit/f2597d65842009116abeb8f5ddd564be815b04cd

But I still get the error at startup:
```
SC FFT global init: cosTable initialised.
init_OSC
empty
compiling class library...
    NumPrimitives = 586
    compiling dir: '/usr/local/share/SuperCollider/SCClassLibrary'
    compiling dir: '/usr/local/share/SuperCollider/Extensions'
    pass 1 done
ERROR: Class extension for nonexistent class 'Document'
     In file:'/deprecated/3.7/deprecated-3.7.sc'
    numentries = 814835 / 11167620 = 0.073
    4990 method selectors, 2238 classes
    method table size 6191868 bytes, big table size 44670480
    Number of Symbols 11653
    Byte Code Size 351432
    compiled 316 files in 2.56 seconds
compile done
ERROR: A primitive was not bound. 0 585
Instance of Method {    (0xb61d20, gc=01, fmt=00, flg=11, set=04)
  instance variables [15]
    raw1 : Float 0.000000   00000000 0038000C
    raw2 : Float 0.000000   00000200 02010002
    code : instance of Int8Array (0xb61ee0, size=4, set=3)
    selectors : nil
    constants : nil
    prototypeFrame : instance of Array (0xb61e60, size=2, set=3)
    context : nil
    argNames : instance of SymbolArray (0xb61de0, size=2, set=3)
    varNames : nil
    sourceCode : nil
    ownerClass : class Meta_QtGUI (0xb59220)
    name : Symbol 'prSetStyle'
    primitiveName : Symbol '_Qt_SetStyle'
    filenameSymbol : Symbol
'/usr/local/share/SuperCollider/SCClassLibrary/Common/GUI/Base/QtGUI.sc'
    charPos : Integer 3196
}
WARNING: Qt style initialisation failed
ERROR: A primitive was not bound. 0 585
Instance of Method {    (0xb41360, gc=01, fmt=00, flg=11, set=04)
  instance variables [15]
    raw1 : Float 0.000000   00000000 0038000C
    raw2 : Float 0.000000   00000200 02010002
    code : instance of Int8Array (0xb41520, size=4, set=3)
    selectors : nil
    constants : nil
    prototypeFrame : instance of Array (0xb414a0, size=2, set=3)
    context : nil
    argNames : instance of SymbolArray (0xb41420, size=2, set=3)
    varNames : nil
    sourceCode : nil
    ownerClass : class Meta_Font (0xb3d860)
    name : Symbol 'prDefaultFamilyForStyle'
    primitiveName : Symbol '_QFont_DefaultFamilyForStyle'
    filenameSymbol : Symbol
'/usr/local/share/SuperCollider/SCClassLibrary/Common/GUI/Base/QFont.sc'
    charPos : Integer 1909
}
WARNING: Qt font initialisation failed
LID: event loop started
Class tree inited in 0.15 seconds
Cleaning up temp synthdefs...
ERROR: A primitive was not bound. 0 585
Instance of Method {    (0xb61260, gc=01, fmt=00, flg=11, set=04)
  instance variables [15]
    raw1 : Float 0.000000   00000000 0030000C
    raw2 : Float 0.000000   00000100 01000001
    code : instance of Int8Array (0xb61320, size=4, set=3)
    selectors : nil
    constants : nil
    prototypeFrame : instance of Array (0xb6787ea0, size=1, set=3)
    context : nil
    argNames : instance of SymbolArray (0xb6787f20, size=1, set=3)
    varNames : nil
    sourceCode : nil
    ownerClass : class Meta_QtGUI (0xb59220)
    name : Symbol 'palette'
    primitiveName : Symbol '_Qt_GlobalPalette'
    filenameSymbol : Symbol
'/usr/local/share/SuperCollider/SCClassLibrary/Common/GUI/Base/QtGUI.sc'
    charPos : Integer 2627
}
ERROR: Primitive 'none' failed.
Failed.
RECEIVER:
   nil
CALL STACK:
    MethodError:reportError   0x222bd0
        arg this = <instance of PrimitiveFailedError>
    Nil:handleError   0xdd1c50
        arg this = nil
        arg error = <instance of PrimitiveFailedError>
    Thread:handleError   0x20c0d0
        arg this = <instance of Thread>
        arg error = <instance of PrimitiveFailedError>
    Object:throw   0x24b3b0
        arg this = <instance of PrimitiveFailedError>
    Object:primitiveFailed   0xdecc90
        arg this = nil
    < closed FunctionDef >  (no arguments or variables)
    Function:doOnStartUp   0xe5dc50
        arg this = <instance of Function>
    ArrayedCollection:do   0x353a90
        arg this = [*6]
        arg function = <instance of Function>
        var i = 5
    List:do   0x353c70
        arg this = <instance of List>
        arg function = <instance of Function>
    Meta_StartUp:run   0x355230
        arg this = <instance of Meta_StartUp>
    Main:startup   0x38b810
        arg this = <instance of Main>
        var didWarnOverwrite = false
^^ The preceding error dump is for ERROR: Primitive 'none' failed.
Failed.
RECEIVER: nil
```

### PR DESCRIPTION
**Pull request by Apteryks**
Purpose and Motivation
----------------------
Fixes #1209

Types of changes
----------------
- Bug fix (non-breaking change which fixes an issue)

Checklist
---------
- [x] All previous tests are passing
- [x] This PR is ready for review

### CODE REVIEW

File: SCClassLibrary/System/System.sc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+System {
+
+	*hasQt {
```
Comment (brianlheim) [inline SCClassLibrary/System/System.sc]: please relocate this to Platform

File: lang/LangPrimSource/PyrSystemPrim.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+such as library support.
+
+*/
+
+#include <PyrErrors.h>
+#include <PyrPrimitive.h>
+#include <PyrSlot.h>
+#include <VMGlobals.h>
+
+int prSystem_hasQt(struct VMGlobals *g, int numArgsPushed)
```
Comment (brianlheim) [inline lang/LangPrimSource/PyrSystemPrim.cpp]: please relocate this to PyrPlatformPrim.cpp

**Comment by Apteryks** [comment]
IIUC, the inherited ^this.primitiveFailed method is only valid for instantiated objects, correct?

**Comment by brianlheim** [comment]
no, you can and should use it in both class and instance methods.

**Comment by Apteryks** [comment]
@brianlheim  I asked because this kind of code in Platform.sc left me wondering:
``` sclang
	systemAppSupportDir {
		_Platform_systemAppSupportDir
		^this.primitiveFailed
	}
	*systemAppSupportDir { ^thisProcess.platform.systemAppSupportDir }
```
I'm still unsure what the class method (*) version is for, or why we couldn't have only the earlier definition as a class method? Sorry for all the questions, this is all rather new to me! Reworked patch will follow shortly.

**Comment by Apteryks** [comment]
@brianlheim I've made the requested changes.

**Comment by brianlheim** [comment]
@Apteryks thanks, will review soon. sorry for the delay

**Comment by snappizz** [comment]
@Apteryks this looks pretty good to me, but there's a conflict now (as a result of the delay -- sorry about that). would you be okay with rebasing this on a newer `develop`? if you don't have time for that, let me know and one of us can take care of it.

**Pull request merged**